### PR TITLE
updated thresholds/values to work with PROJ 9.2

### PR DESCRIPTION
**Pull request by kirillrdy**
## Rationale

fixes https://github.com/SciTools/cartopy/issues/2145

## Implications

I cant speak for correctness of values
from release log https://proj.org/news.html#release-notes 
it looks like EPSG and ESRI databases got updated so its probably related

### CODE REVIEW

**Comment by greglucas** [comment]
This probably needs a check on PROJ(pyproj) versions within the tests to pass the minimum versions CI as well.

File: lib/cartopy/tests/crs/test_equidistant_conic.py [hunk 1]
```diff
@@ -23,12 +24,20 @@ def test_default(self):
                       'y_0=0.0', 'lat_1=20.0', 'lat_2=50.0'}
         check_proj_params('eqdc', eqdc, other_args)
 
-        assert_almost_equal(np.array(eqdc.x_limits),
-                            (-22784919.35600352, 22784919.35600352),
-                            decimal=7)
-        assert_almost_equal(np.array(eqdc.y_limits),
-                            (-10001965.729313632, 17558791.85156368),
-                            decimal=7)
+        if pyproj.__proj_version__ >= '9.2.0':
```
Comment (greglucas) [inline lib/cartopy/tests/crs/test_equidistant_conic.py:27]: What are your thoughts about changing the number of decimal places we compare to here? It seems like if we can compare to 2-3+ decimal places that should be good enough for Cartopy's purposes. We are really relying on proj for all the "official" transformations.

File: lib/cartopy/tests/crs/test_equidistant_conic.py [hunk 2]
```diff
@@ -23,12 +24,20 @@ def test_default(self):
...
+            assert_almost_equal(np.array(eqdc.x_limits),
+                                (-22784919.3559981,  22784919.3559981),
+                                decimal=7)
+            assert_almost_equal(np.array(eqdc.y_limits),
+                                (-10001965.72931272,  17558791.85157471),
+                                decimal=7)
```
Comment (greglucas) [inline lib/cartopy/tests/crs/test_equidistant_conic.py:33]: I would restructure these blocks to something like this to avoid the duplicated `assert_almost_equal`s.
```suggestion
        expected_x = ...
        expected_y = ...
        if pyproj.__proj_version__ >= '9.2.0':
            expected_x = ...
            expected_y = ...
        assert_array_equal(eqdc.x_limits, expected_x)
        assert_array_equal(eqdc.y_limits, expected_y)
```

**Review by greglucas — COMMENTED** [review]
Looks good, just a few minor comments to consider.

File: lib/cartopy/tests/crs/test_equidistant_conic.py (hunk 1 above)
Comment (kirillrdy) [reply lib/cartopy/tests/crs/test_equidistant_conic.py:29]: I don't have any strong opinion about number of decimals. I do find it strange that cartopy supports different versions of proj that would lead to very different results (eg boundary of UK) 

I would either not test values from proj at all, since they change between versions OR only support one version of proj that is considered "correct", most likely latest version of proj.

that's just my 2 cents but no need to change anything for this PR

File: lib/cartopy/tests/crs/test_equidistant_conic.py (hunk 2 above)
Comment (kirillrdy) [reply lib/cartopy/tests/crs/test_equidistant_conic.py]: done

**Comment by kirillrdy** [comment]
i see tests are failing, but passing in my setup, I have an other look

**Comment by kirillrdy** [comment]
should be fixed now

File: lib/cartopy/tests/crs/test_equidistant_conic.py (hunk 1 above)
Comment (greglucas) [reply lib/cartopy/tests/crs/test_equidistant_conic.py:29]: I agree with everything you say here :)

**Review by greglucas — APPROVED** [review]
